### PR TITLE
ref(hc): Remove get_orgs()

### PR DESCRIPTION
**Pull request by corps**
user.get_orgs() is problematic, because the Organization objects associated to a user via OrganizationMember can only ever be *per region*, and not contain all organizations a user belongs to globally.

This PR attempts to use separate means of getting the same question answered but with more specific implementations, so we can extract the `user.get_orgs()` which is ambiguous. 

For instance, `Organization.get_for_user_ids` can only mean *within a region scope*.  but `user_service.get_organizations` will include global organizations.

### CODE REVIEW

File: src/sentry/api/bases/sentryapps.py [hunk 1]
```diff
@@ -77,17 +80,17 @@ class SentryAppsPermission(SentryPermission):
         "POST": ("org:write", "org:admin"),
     }
 
-    def has_object_permission(self, request: Request, view, organization):
```
Comment (corps) [inline src/sentry/api/bases/sentryapps.py:80]: SentryApps will be control silo, which means they won't have direct access to organizations.  the context here represents an RPC lookup of org details.

File: src/sentry/api/endpoints/user_notification_fine_tuning.py [hunk 2]
```diff
@@ -89,28 +93,11 @@ def put(self, request: Request, user, notification_type) -> Response:
 
         # Make sure that the IDs we are going to update are a subset of the
         # user's list of organizations or projects.
-        parents = (
```
Comment (corps) [inline src/sentry/api/endpoints/user_notification_fine_tuning.py:92]: Requiring validation of project is and organization ids forces a fan out across all regions for a user.  In reality, these values are not foreign keys.  We also frequently "leave behind" notification settings without clearing them in all cases, so letting arbitrary values from a user isn't particularly problematic.  Having notification settings for a project or org you don't belong to doesn't grant any access or necessarily grant any notifications.

File: src/sentry/api/endpoints/user_organizations.py [hunk 3]
```diff
@@ -1,17 +1,45 @@
...
+from sentry.api.bases.user import UserPermission
+from sentry.api.exceptions import ResourceDoesNotExist
 from sentry.api.paginator import OffsetPaginator
 from sentry.api.serializers import serialize
+from sentry.models import Organization, User
+from sentry.services.hybrid_cloud.user import RpcUser
+from sentry.services.hybrid_cloud.user.service import user_service
+
+
+@region_silo_endpoint
```
Comment (corps) [inline src/sentry/api/endpoints/user_organizations.py:18]: Swapping this to a region endpoint.  In the future, we plan to have the frontend query a user's organization settings per non single tenant region they belong to.  Pushing this to the frontend makes sense in the long term.

**Review by dashed — APPROVED** [review]

File: src/sentry/features/helpers.py [hunk 4]
```diff
@@ -18,16 +18,11 @@ def any_organization_has_feature(
     return any([features.has(feature, organization, **kwargs) for organization in organizations])
 
 
-def requires_feature(
-    feature: str, any_org: Optional[bool] = None
-) -> Callable[[EndpointFunc], EndpointFunc]:
+def requires_feature(feature: str) -> Callable[[EndpointFunc], EndpointFunc]:
```
Comment (dashed) [inline src/sentry/features/helpers.py:21]: @corps We can just delete this function as it's no longer used anywhere.

Comment (corps) [reply src/sentry/features/helpers.py:21]: Nice, thanks!

**Pull request merged**